src/core/util/tools.py

```
@@ -23,19 +23,30 @@
-                           universal_newlines=True, shell=True, bufsize=1)
-    info = ""
-    # 实时输出
-    while True:
-        line = cmd.stderr.readline().strip()
-        Constants.log.info(f"[shell] {line}")
-        info += line
-
-        if line == "" or subprocess.Popen.poll(cmd) == 0:  # 判断子进程是否结束
-            break
+
+    with subprocess.Popen(shell, stdin=subprocess.PIPE, stderr=subprocess.PIPE,
```

Ruff (0.13.1)

`subprocess` call with `shell=True` identified, security issue

(S602)